src/provider/mod.rs

```diff
@@ -109,17 +161,30 @@ where
     }
 
     /// Retrieves the log proof for an L2 to L1 transaction.
+    /// # Parameters
+    ///
+    /// - `tx_hash`: hash of the L2 transaction the L2 to L1 log was produced in.
+    /// - `l2_to_l1_log_index`: Optional: The index of the L2 to L1 log in the transaction.
+    ///
+    /// # Returns
+    /// The L2 to L1 log proof.
     fn get_l2_to_l1_log_proof(
         &self,
         tx_hash: B256,
-        index: Option<usize>,
+        l2_to_l1_log_index: Option<usize>,
     ) -> ProviderCall<T, (B256, Option<usize>), Option<L2ToL1LogProof>> {
         self.client()
-            .request("zks_getL2ToL1LogProof", (tx_hash, index))
+            .request("zks_getL2ToL1LogProof", (tx_hash, l2_to_l1_log_index))
             .into()
     }
 
     /// Retrieves details for a given block.
+    /// # Parameters
+    ///
+    /// - `block_number`: The number of the block.
+    ///
+    /// # Returns
+    /// The details of the L2 block.
```

Comment:
Probably here we can just mention `Retrieves details for a given L2 block` to make it more succinct.